src/main/java/edu/usf/sas/pal/muser/manager/MuserMusicDataAnalysisManager.java

```diff
@@ -62,27 +90,79 @@ private void analyzeAllUsersMusicData() {
         for (QueryDocumentSnapshot doc : allUserIds) {
             processUserByIdForPlayerEvent(doc.getId());
             processUserByIdForUiEvent(doc.getId());
+            processUserByIdForDeviceInfo(doc.getId());
         }
     }
 
-    private void processUserByIdForPlayerEvent(String userId) {
+    private void filterUsersDataByDataRange(long millisStart, long millisEnd) {
+        List<QueryDocumentSnapshot> allUserIds = firebaseReader.getAllUserIds();
+        for (QueryDocumentSnapshot doc : allUserIds) {
+            processUserByDateRangeForPlayerEvent(doc.getId(), millisStart, millisEnd);
+            processUserByIdDateRangeForUiEvent(doc.getId(), millisStart, millisEnd);
+            processUserByIdDateRangeForDeviceInfo(doc.getId(), millisStart, millisEnd);
+        }
+    }
+
+    private void filterParticularUsersDataByDataRange(long millisStart, long millisEnd, String userId) {
+        processUserByDateRangeForPlayerEvent(userId, millisStart, millisEnd);
+        processUserByIdDateRangeForUiEvent(userId, millisStart, millisEnd);
+        processUserByIdDateRangeForDeviceInfo(userId, millisStart, millisEnd);
+    }
 
-        List<QueryDocumentSnapshot> userPlayerEventInfoById = new ArrayList<>(firebaseReader.getAllUserPlayerEventInfoById(userId));
+    private void processUserByIdForPlayerEvent(String userId) {
+        //get data of Player Events collection based on user id
```

Comment:
Nit - in typical Java formatting you have a space between the `//` and words - `// Get data...`